src/linux/address.rs

```diff
@@ -7,6 +7,28 @@ pub trait Ipv4AddrExt {
     fn from_address(sock: sockaddr) -> Self;
 }
 
+// Ext suffix is in case a dependency on an external MAC address crate is ever pulled in
+// Also matches above naming this way
+pub trait MACAddrExt {
+    fn to_address(&self) -> sockaddr;
+    fn from_address(sock: sockaddr) -> Self;
+}
+
+pub struct MACAddr {
```

Comment:
Maybe `MacAddr` is a better name for this struct like the standard library which has `TcpListener` instead of `TCPListener`.